### PR TITLE
Add Filter1XXHeaderStatus.

### PR DESCRIPTION
**Pull request by KBaichoo**
Signed-off-by: Kevin Baichoo <kbaichoo@google.com>

Commit Message: Use different status code for 1xxHeaders.
Additional Description:  This is mostly `s/FilterHeadersStatus/Filter1xxHeadersStatus/`. Prior we could return additional FilterHeaderStatus status codes but those weren't actually supported in the code for 1xx headers. 
Risk Level: low
Testing: ran existing tests
Docs Changes: included
Release Notes: included
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by KBaichoo** [comment]
/assign @alyssawilk 

Can you take a look since we talked about this offline you might have the most context. Thank you!

File: changelogs/current.yaml [hunk 1]
```diff
@@ -26,6 +26,9 @@ minor_behavior_changes:
 - area: cache_filter
   change: |
     add a completion callback to updateHeaders interface. Any external cache implementations will need to update to match this new interface. See changes to simple_http_cache in PR#23666 for example.
+- area: http filters
+  change: |
+    change ``StreamEncoderFilter::encode1xxHeaders`` to use its own enum class ``Http::Filter1xxHeadersStatus``. Prior we shared the same enum class for general headers, but the implementation did not support most of them.
```
Comment (alyssawilk) [inline changelogs/current.yaml:31]: s/Prior/Previously

File: envoy/http/filter.h [hunk 2]
```diff
@@ -105,6 +105,19 @@ enum class FilterHeadersStatus {
...
 
+/**
+ * Return codes for encode on informative headers filter invocations.
+ */
+enum class Filter1xxHeadersStatus {
+  // Continue filter chain iteration.
+  Continue,
+  // Do not iterate for informative headers on any of the remaining filters in the chain.
+  //
+  // Returning FilterDataStatus::Continue from encodeData() or calling
```
Comment (alyssawilk) [inline envoy/http/filter.h:116]: FilterDataStatus::Continue? copy paste error?

**Review by alyssawilk — COMMENTED** [review]
LG over all, again nice catch on this one!
/wait

File: envoy/http/filter.h (hunk 2 above)
Comment (KBaichoo) [reply envoy/http/filter.h:116]: This unfortunately is the existing behavior. Messaged out of band , happy to handle this in a follow up to keep this as a clean refactor.

**Review by alyssawilk — APPROVED** [review]

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**